online-shop/src/app/state/reducers/login.reducer.ts

```diff
@@ -0,0 +1,36 @@
+import {UserCredentials} from "../../interfaces/UserCredentials";
+import {createReducer, on} from "@ngrx/store";
+import {login, loginFailure, loginSuccessful, verifyAdmin} from "../actions/login.actions";
+import {User} from "../../interfaces/User";
+
+export interface LoginState {
+  loggedInUser: User | undefined;
+  error: string | null;
+  status: 'pending' | 'loading' | 'error' | 'success';
```

Comment:
This can be extracted in a separate type and reused for all. But normally, it's mostly a simple boolean only, named loading.